pkg/vm/vm.go

```diff
@@ -1492,10 +1496,24 @@ func cloneIfStruct(item StackItem) StackItem {
 	}
 }
 
-func makeArrayOfFalses(n int) []StackItem {
+func makeArrayOfType(n int, typ StackItemType) []StackItem {
 	items := make([]StackItem, n)
+	var c func() StackItem
```

Comment:
I don't think this situation is worth creating this callback, you can check for `typ` validity once and then use a switch inside a range loop. Alternatively we also have a `Dup()` method that can be used.

Reply:
Fixed.